lib/actions/close.js

```diff
@@ -11,20 +11,26 @@ class Close extends Action {
     super('close')
     this.supportedEvents = [
       'pull_request.*',
-      'issues.*'
+      'issues.*',
+      'schedule.repository'
     ]
   }
 
   // there is nothing to do
   async beforeValidate () {}
 
   async afterValidate (context, settings, name, results) {
-    const payload = this.getPayload(context)
-    const issueNumber = payload.number
-
-    return closeIssue(
-      context,
-      issueNumber
+    let scheduleResults = results && results.validationSuites && results.validationSuites[0].schedule
```

Comment:
have you tested these with non `schedule.repository` flows? I don't remember if normal flows have `.schedule` properties thus this might fail

Reply:
yes I have. For non schedule flows the items variable below is set to getPayload(context) instead of the schedule results since schedule results will be undefined.

Reply:
This logic is similar to the logic in the comment action.

Reply:
ah ok, i see it now.

Reply:
There were already some test cases for the single issue. I also added a test case for scheduler.